### DISCUSSION TITLE
MutationObserver feature support for node module's global object or specific objects.

### DISCUSSION THREAD

**Discussion by ganeshkbhat** [nodejs/help#4706] [Q&A]
### What is the problem this feature will solve?

 Compatibility with Nodejs? MutationObserver will support Observers on global object or specific objects.

### What is the feature you are proposing to solve the problem?

Can I use this [MutationObserver](https://github.com/webmodules/mutation-observer) for node module's objects declaration tracking and observing declared objects on current nodejs version 18.14.2 LTS, and 19.7.0 Current? Any polyfill also should do.

[https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver)

### What alternatives have you considered?

None. Recommendation for support of similar feature; alternatively.

**Comment by meixg** [comment]
I think the [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) is used to observe DOM, what you want is probably [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) or [defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty).

**Comment by bnoordhuis** [comment]
@meixg is right. I'll convert this to a discussion.

**Comment by ganeshkbhat** [comment]
@meixg @bnoordhuis Do you think considering this for global object observer is a good thought? I do think so. However, this is a memory leak risk.

**Reply by bnoordhuis** [reply]
There's no DOM in node so what would be the point?
